src/UKHO.ERPFacade.API/Services/S57Service.cs

```diff
@@ -0,0 +1,99 @@
+        {
+            var eventData = JsonConvert.DeserializeObject<EncEventPayload>(encEventJson.ToString());
+
+            if (IsAioCell(eventData.Data.Products.Select(x => x.ProductName).ToList()))
+            {
+                _logger.LogInformation(EventIds.NoProcessingOfNewEncContentPublishedEventForAioCells.ToEventId(), "The enccontentpublished event will not be processed for Aio cells.");
+                return;
+            }
+
+            EncEventEntity encEventEntity = new()
+            {
+                RowKey = Guid.NewGuid().ToString(),
+                PartitionKey = Guid.NewGuid().ToString(),
+                Timestamp = DateTime.UtcNow,
+                CorrelationId = eventData.Data.CorrelationId,
+                RequestDateTime = null
+            };
+
+            _logger.LogInformation(EventIds.AddingEntryForEncContentPublishedEventInAzureTable.ToEventId(), "Adding/Updating entry for enccontentpublished event in azure table.");
+            await _azureTableReaderWriter.UpsertEntity(eventData.Data.CorrelationId, Constants.S57EventTableName, encEventEntity);
+
+            _logger.LogInformation(EventIds.UploadEncContentPublishedEventInAzureBlobStarted.ToEventId(), "Uploading enccontentpublished event payload in blob storage.");
+            await _azureBlobEventWriter.UploadEvent(encEventJson.ToString(), Constants.S57EventContainerName, eventData.Data.CorrelationId + '/' + Constants.S57EncEventFileName);
+            _logger.LogInformation(EventIds.UploadEncContentPublishedEventInAzureBlobCompleted.ToEventId(), "The enccontentpublished event payload is uploaded in blob storage successfully.");
+
+            var sapPayload = _encContentSapMessageBuilder.BuildSapMessageXml(JsonConvert.DeserializeObject<EncEventPayload>(encEventJson.ToString()));
+
+            _logger.LogInformation(EventIds.UploadSapXmlPayloadInAzureBlobStarted.ToEventId(), "Uploading the SAP XML payload in blob storage.");
+            await _azureBlobEventWriter.UploadEvent(sapPayload.ToIndentedString(), Constants.S57EventContainerName, eventData.Data.CorrelationId + '/' + Constants.SapXmlPayloadFileName);
+            _logger.LogInformation(EventIds.UploadSapXmlPayloadInAzureBlobCompleted.ToEventId(), "SAP XML payload is uploaded in blob storage successfully.");
+
+            var response = await _sapClient.PostEventData(sapPayload, _sapConfig.Value.SapEndpointForEncEvent, _sapConfig.Value.SapServiceOperationForEncEvent, _sapConfig.Value.SapUsernameForEncEvent, _sapConfig.Value.SapPasswordForEncEvent);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                throw new ERPFacadeException(EventIds.RequestToSapFailed.ToEventId(), $"An error occurred while sending a request to SAP. | {response.StatusCode}");
+            }
+            _logger.LogInformation(EventIds.EncUpdateSentToSap.ToEventId(), $"ENC update has been sent to SAP successfully. | {response.StatusCode}");
```

Comment:
Snyk throws "Unsanitized input from an HTTP parameter flows into LogInformation, where it is used as log input. This may result in a Logs Forging vulnerability." 
![ForgingVulnerability](https://github.com/user-attachments/assets/901f3c8c-c84e-4801-9af6-8855c016f57e)